truckersmp-cli

```diff
@@ -1,512 +1,547 @@
+    steamcmdurl = "https://steamcdn-a.akamaihd.net/client/installer/steamcmd_linux.tar.gz"
+    issueurl = "https://github.com/lhark/truckersmp-cli/issues"
+
+class AppId:
+    """ steam AppIds """
+    game = {
+        "ats"         : 270880,        # https://steamdb.info/app/270880/
+        "ets2"        : 227300,        # https://steamdb.info/app/227300/
+    }
+    proton = {
+        "4.2"         : 1054830,       # https://steamdb.info/app/1054830/
+        "3.16 Beta"   : 996510,        # https://steamdb.info/app/996510/
+        "3.16"        : 961940,        # https://steamdb.info/app/961940/
+        "3.7 Beta"    : 930400,        # https://steamdb.info/app/930400/
+        "3.7"         : 858280,        # https://steamdb.info/app/858280/
+    }
+    default_proton_ver = "4.2"
+
+class Dir:
+    """ directories """
+    XDG_CACHE_HOME = os.getenv("XDG_CACHE_HOME",
+      os.path.join(os.path.expanduser("~"), ".cache"))
+    XDG_DATA_HOME = os.getenv("XDG_DATA_HOME",
+      os.path.join(os.path.expanduser("~"), ".local/share"))
+    default_gamedir = {
+        "ats" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/American Truck Simulator/data"),
+        "ets2" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/Euro Truck Simulator 2/data"),
+    }
+    default_prefixdir = {
+        "ats" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/American Truck Simulator/prefix"),
+        "ets2" : os.path.join(XDG_DATA_HOME,
+          "truckersmp-cli/Euro Truck Simulator 2/prefix"),
+    }
+    default_moddir = os.path.join(XDG_DATA_HOME, "truckersmp-cli/TruckersMP")
+    default_protondir = os.path.join(XDG_DATA_HOME, "truckersmp-cli/Proton")
+    scriptdir = os.path.dirname(os.path.realpath(__file__))
+    steamcmddir = os.path.join(XDG_CACHE_HOME, "truckersmp-cli/steamcmd")
+
+
+def printdbg(msg, end = "\n"):
+    if args.verbose:
+        print(msg, end = end)
+
+def printerr(msg, end = "\n"):
+    print(msg, end = end, file = sys.stderr)
+
+def download_files(host, files_to_download):
+    try:
+        conn = http.client.HTTPSConnection(host)
+    except Exception as e:
+        printerr("Failed to connect to {}: {}".format(host, e))
+        return False
+    try:
+        while len(files_to_download) > 0:
+            path, dest, md5 = files_to_download[0]
+            md5hash = hashlib.md5()
+            bufsize = md5hash.block_size * 256
+            name = os.path.basename(path)
+            destdir = os.path.dirname(dest)
+            printdbg(
+              "Downloading file https://{}{} to {}".format(host, path, destdir))
+
+            # make file hierarchy
+            os.makedirs(destdir, exist_ok = True)
+
+            # download file
+            conn.request("GET", path, headers = {"Connection" : "keep-alive"})
+            res = conn.getresponse()
+
+            if res.status != 200:
+                printerr("Server response code is not 200.")
```

Comment:
Printing the actual status code would be more useful. Something like `"Error: server {} responded with status code {}".format(host, res.status)`